tools/v8_gypfiles/v8.gyp

```diff
@@ -41,6 +41,15 @@
         'AdditionalOptions': ['/utf-8']
       }
     },
+    # Hide symbols that are not explicitly exported with V8_EXPORT.
+    # TODO(joyeecheung): enable it on other platforms. Currently gcc times out
+    # or run out of memory with -fvisibility=hidden on some machines in the CI.
+    'xcode_settings': {
+      'GCC_SYMBOLS_PRIVATE_EXTERN': 'YES',  # -fvisibility=hidden
+    },
+    'defines': [
+      'BUILDING_V8_SHARED',  # Make V8_EXPORT visible.
```

Comment:
https://github.com/nodejs/node/blob/5a868b5b30e551d375c73d65013883e6078bba84/common.gypi#L478-L479 non-blocking: `common.gypi` probably needs to be consolidated

Reply:
I think for addons, `common.gypi` should define `USING_V8_SHARED` instead, then in `node.gypi` we need to undefine it/override it to `BUILDING_V8_SHARED` when building libnode, though that might be a bit challenging to get right with the messy gyp config inheritance all over the place..for now there should probably not even be any difference on Windows (`USING_V8_SHARED` only affects Windows AFAICT, but currently both the addons and the Node.js build have `BUILDING_V8_SHARED` on Windows before and after this PR, even though it's technically not quite right for the addons to do that). We can look into it in a follow up.

Reply:
yeah, agreed that addons should have `USING_V8_SHARED` defined. Right now, it is defined in node-gyp. So here in common.gypi, it may be removed in a follow-up.

https://github.com/nodejs/node-gyp/blob/b899faed56270d3d8496da7576b5750b264c2c21/addon.gypi#L36